olive/cli/finetune.py

```diff
@@ -0,0 +1,267 @@
+
+    def parse_training_args(self) -> Dict:
+        if not self.unknown_args:
+            return {}
+
+        from transformers import HfArgumentParser, TrainingArguments
+
+        arg_keys = {el[2:] for el in self.unknown_args if el.startswith("--")}
+        parser = HfArgumentParser(TrainingArguments)
+        # output_dir is required by the parser
+        training_args = parser.parse_args(
+            [*(["--output_dir", "dummy"] if "output_dir" not in arg_keys else []), *self.unknown_args]
+        )
+
+        return {k: v for k, v in vars(training_args).items() if k in arg_keys}
+
+    def get_run_config(self) -> Dict:
+        load_key = ("data_configs", 0, "load_dataset_config")
+        preprocess_key = ("data_configs", 0, "pre_process_data_config")
+        finetune_key = ("passes", "f")
+        to_replace = [
+            (("input_model", "model_path"), self.args.model_name_or_path),
+            ((*load_key, "data_name"), self.args.data_name),
+            ((*load_key, "split"), self.args.train_split),
+            (
+                (*load_key, "data_files"),
+                self.args.data_files.split(",") if self.args.data_files else None,
+            ),
+            ((*preprocess_key, "text_cols"), self.args.text_field),
+            ((*preprocess_key, "text_template"), self.args.text_template),
+            ((*preprocess_key, "max_seq_len"), self.args.max_seq_len),
+            ((*finetune_key, "type"), self.args.method),
+            ((*finetune_key, "torch_dtype"), self.args.torch_dtype),
+            ((*finetune_key, "training_args"), self.parse_training_args()),
+            ((*finetune_key, "lora_r"), self.args.lora_r),
+            ((*finetune_key, "lora_alpha"), self.args.lora_alpha),
+            (("passes", "o", "float16"), self.args.precision == "float16"),
+            # make the mapping of precisions better
+            (("passes", "m", "precision"), "fp16" if self.args.precision == "float16" else "fp32"),
+            (("clean_cache",), self.args.clean),
+        ]
+        if self.args.trust_remote_code:
+            to_replace.append((("input_model", "load_kwargs", "trust_remote_code"), True))
+        if self.args.method == "lora":
+            to_replace.append(((*finetune_key, "target_modules"), self.args.target_modules.split(",")))
+
+        config = deepcopy(TEMPLATE)
+        for keys, value in to_replace:
+            if value is None:
+                continue
+            set_nested_dict_value(config, keys, value)
+
+        if self.args.eval_split:
+            eval_data_config = deepcopy(config["data_configs"][0])
+            eval_data_config["name"] = "eval_data"
+            eval_data_config["load_dataset_config"]["split"] = self.args.eval_split
+            config["data_configs"].append(eval_data_config)
+            config["passes"]["f"]["eval_data_config"] = "eval_data"
+
+        return config
+
+
+TEMPLATE = {
```

Comment:
How about move this kind of template under "auto_optimzier" ? It will be helpful for coming auto optimizer with well-know workflow templates.
![image](https://github.com/user-attachments/assets/cbbd48f7-6bb8-4e86-a27c-4bae781bf188)

Reply:
Can I do it as a follow up PR? There is no use case for the auto optimizer currently so I don't have anything to test this for?

Reply:
Sure.